app/src/main/java/cloudcity/LoggingServiceExtensions.java

```diff
@@ -183,45 +180,10 @@ private static NetworkDataModel getCloudCityData() {
         /* Convert to km/h */
         dataModel.setSpeed(location.getSpeed() * 3.6);
 
-        dataModel.setCellData(getCellInfoModel(category, currentCell));
+        dataModel.setCellData(
+                CellUtil.getCellInfoModel(currentCell)
+        );
 
         return dataModel;
     }
-
-
-
-    private static @NonNull CellInfoModel getCellInfoModel(String category, CellInformation currentCell) {
-        CellInfoModel cellInfoModel = new CellInfoModel();
-
-        if (Objects.equals(category, "CDMA") || Objects.equals(category, "GSM")) {
-            /* No information in 2G and 3G set dummy data. */
-            cellInfoModel.setDummy(1);
-        } else {
-            /* Real data available for all other network types. */
-            if (currentCell instanceof GSMInformation) {
-                GSMInformation gsmCell = (GSMInformation) currentCell;
-                // Ok so bands is technically the ARFCN
-                String bandsString = gsmCell.getBands();
-                int arfcn = Integer.parseInt(bandsString);
-                cellInfoModel.setEarfcn(arfcn);
-                cellInfoModel.setPci(currentCell.getPci());
-            }
```

Comment:
apart from 'breaking' this after the first upstream sync and making it work only for GSM networks, which are literally caught in the `if()` above (thus this code would never execute), the EARFCN was either not working, or the OMNT team changed the format of the `bands` and how they return them.

If this code worked (apart from the GSMInformation limit), they used to send `bands` as a simple numeric string, e.g. `"3"`. But now, they send something that seems like a list (which will also give this code some problems), but at least it works fine for these two cases:
1) having no SIM, which is when the bands arrive as `"[]"`
2) having only one SIM, which is when the bands arrive as `"[3]"` or `"[20]"`
and the new sanitized string followed by parsing (as implemented in `CellUtil`) correctly captures both of these situations.

this is just a clarification comment, there's nothing to be done about it now :)